src/test/java/org/broadinstitute/hellbender/utils/nio/GcsNioIntegrationTest.java

```diff
@@ -47,7 +47,8 @@ public void openPublicFile() throws IOException {
     @Test(groups={"bucket"})
     public void writePrivateFile() throws IOException {
         try {
-            final String dest = getGCPTestStaging() + "GcsNioIntegrationTest-writePrivateFile-test" + new Random().nextInt();
+            final String dest = BucketUtils.randomRemotePath(
```

Comment:
why not use `BucketUtils.getTempFilePath()`?  That will autodelete as well.

Reply:
@lbergelson God idea. I also changed the ReadUtils test (the other place in this PR that I had modified) to use BucketUtils.getTempFilePath() as well, which in turn allowed me to remove some of the local file deletion code and a try/finally block from that file.

Also, I notice that ".md5" is not it included in the list of file extensions that BucketUtils.getTempFilePath() auto-deletes. So I added that. Can  you please take a look the new commit ? Thx.